src/api/VoteMonitor.Api.Form/Mappers/HierarchicalMapper.cs

```diff
@@ -0,0 +1,70 @@
+﻿using AutoMapper;
+using System.Collections.Generic;
+using System.Linq;
+using VoteMonitor.Entities;
+
+namespace VoteMonitor.Api.Form.Mappers
+{
+    /// <summary>
+    /// Base class using Template Design Pattern in order to facilitate the mapping of hierarchical objects.
+    /// </summary>
+    public abstract class HierarchicalMapper<TEntity, TDto, TChildEntity, TChildDto> : UpdateOrCreateEntityMapper<TEntity, TDto>
```

Comment:
This will most likely won't be instantiated by DI as it is abstract

Reply:
I suggest to keep them separately for the reason they don't share anything right now (ie hierarchical mapper doesn't use the automapper, ctor only passes it to the other mapper)

Reply:
Maybe I'm wrong but in case if we're gonna separate them, we'll still need the automapper in the HierarchicalMapper in order to map the current entity and we'll have duplicated code(the one from UpdateOrCreateEntityMapper).
We cannot use composition here(to inject a IUpdateOrCreateEntityMapper in the HierarchicalMapper) as in that case, the same hierarchical mapper will be injected(as they should use the same interface. Do you have any suggestions for this?